packages/flutter/test/cupertino/refresh_test.dart

```diff
@@ -535,6 +528,94 @@ void main() {
       debugDefaultTargetPlatformOverride = null;
     });
 
+    testWidgets('builder still called when sliver snapped back more than 90%', (WidgetTester tester) async {
+      debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+      refreshIndicator = const Center(child: Text('-1'));
+
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: CustomScrollView(
+            slivers: <Widget>[
+              CupertinoSliverRefreshControl(
+                builder: builder,
+                onRefresh: onRefresh,
+              ),
+              buildAListOfStuff(),
+            ],
+          ),
+        ),
+      );
+
+      await tester.drag(find.text('0'), const Offset(0.0, 150.0));
+      await tester.pump();
+      verify(mockHelper.builder(
+        any,
+        RefreshIndicatorMode.armed,
+        150.0,
+        100.0, // Default value.
+        60.0, // Default value.
+      ));
+      expect(
+        tester.getRect(find.widgetWithText(Center, '-1')),
+        Rect.fromLTRB(0.0, 0.0, 800.0, 150.0),
+      );
+      verify(mockHelper.refreshTask());
+
+      // Rebuilds the sliver with a layout extent now.
+      await tester.pump();
+      // Let it snap back to occupy the indicator's final sliver space only.
+      await tester.pump(const Duration(seconds: 2));
+      verify(mockHelper.builder(
+        any,
+        RefreshIndicatorMode.refresh,
+        60.0,
+        100.0, // Default value.
+        60.0, // Default value.
+      ));
+      expect(
+        tester.getRect(find.widgetWithText(Center, '-1')),
+        Rect.fromLTRB(0.0, 0.0, 800.0, 60.0),
+      );
+      expect(
+        tester.getRect(find.widgetWithText(Center, '0')),
+        Rect.fromLTRB(0.0, 60.0, 800.0, 260.0),
+      );
+
+      refreshCompleter.complete(null);
+      await tester.pump();
+      verify(mockHelper.builder(
+        any,
+        RefreshIndicatorMode.done,
+        60.0,
+        100.0, // Default value.
+        60.0, // Default value.
+      ));
+
+      // Dragging to 5% of refresh control height
+      await tester.drag(find.text('0'), const Offset(0.0, -57.0));
+      await tester.pump();
+
+      // Waiting for refresh control to reach approximately 5% of height
+      await tester.pump(const Duration(milliseconds: 300));
```

Comment:
I'm not sure what it means to both drag up 57px and then also wait 300ms. I'd assume just dragging and pumping would be sufficient

Reply:
This is what happens if I remove line with waiting (601):

```
00:05 +8: UI tests long list builder still called when sliver snapped back more than 90%                
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure object was thrown running a test:
  Expected: 3.0 (±0.4)
  Actual: <30.655658721923828>
```

I'd better increase waiting time to 400ms and remove dragging because it's redundant, test is passing without this step. And I think, that it's a more expected behavior to wait until refresh control collapses than manually dragging it up until it disappears